pages/api/sigil/[address]/tokenId/[tokenId].ts

```diff
@@ -0,0 +1,263 @@
+
+    return data;
+
+  } catch (error) {
+    console.error(`error calling API for Wallet data: ${error}`);
+  }
+}
+
+//Returns processed data for easy use when constucting the NFT
+function processWalletAPIData(data: any): ProcessedWalletData {
+
+  const processedWalletData = {} as ProcessedWalletData;
+
+  try {
+    //WalletData will return an undefined if the owner does not own any Hashes NFTs
+    if (data === undefined) {
+
+      processedWalletData.dao = 0;
+      processedWalletData.non_dao = 0;
+      processedWalletData.votes = 0;
+      processedWalletData.proposals = 0;
+    }
+    else {
+
+      var dao = 0;
+      var non_dao = 0;
+
+      for (let i = 0; i < data.hashes.length; i++) {
+
+        //Adds entry to either DAO or non-DAO increment
+        if (data.hashes[i].type === "DAO") {
+          dao += 1;
+        }
+        else {
+          non_dao += 1;
+        }
+      }
+
+      processedWalletData.dao = dao;
+      processedWalletData.non_dao = non_dao;
+      processedWalletData.votes = data.on_chain_votes;
+
+      if (data.proposals_created === undefined) {
+        processedWalletData.proposals = 0;
+      } else {
+        processedWalletData.proposals = data.proposals_created;
+      }
+    }
+  } catch (error) {
+    console.error(`error transposing wallet data: ${error}`);
+  }
+
+  return processedWalletData;
+}
+
+//////////////////////////////////////////////
+//Helper Functions For Each Sigil Collection//
+//////////////////////////////////////////////
+
+///////////
+//The API//
+///////////
+
+//Returns the json metadata in the Opensea format
+export default async function handler(req: any, res: any) {
+
+    //Sigil Collection Address
+    const { address } = req.query;
+
+    if (typeof(address) !== 'string') {
+      res.status(400).send('address must be a string');
+      return;
+    }
+  
+    if (!isValidAddress(address)) {
+      res.status(400).send('valid (non-ens) wallet address must be provided');
+      return;
+    }
+
+    //////////////////////////////////////////
+    //Try to fetch collection JSON data here//
+    //////////////////////////////////////////
+
+    //This can be deleted if try-fetch collection is built - will be redundant
+    if(!Collections.hasOwnProperty(address)) {
+      res.status(400).send('sigil collection either does not exist or has not been integrated yet');
+      return;
+    }
+
+    //So can this probably
+    const collectionJSONData = Collections[address as keyof typeof Collections];
+
+    //Conforms the address into a hashes collection format
+    const sigilContract = getHashesCollectionContract(address);
+
+    //Sigil Token Id given the collection address
+    const { tokenId } = req.query;
+  
+    if (isNaN(Number(tokenId))) {
+      res.status(400).send('tokenId must be a number');
+      return;
+    }
+
+    //Gets the supply of the collection
+    const getTotalSupply = await sigilContract.totalSupply();
```

Comment:
Awaits here could be wrapped in a try/catch, as they’re async operations and could fail